### ISSUE TITLE
Support named arguments, or generate a better error message

### ISSUE THREAD

**Issue by leonsenft** [dart-lang/angular#1454]
Named method arguments aren't supported in templates. As reported by an external user, this can lead to some very misleading template parsing errors.

For example, the following snippet

```html
<material-input (ngModelChange)="item.copyWith(name: $event)" ...>
```

produces an error like

```sh
[SEVERE] angular on lib/src/item_component.dart:
Parser Error: Missing expected ) at column 22 in [item.copyWith(name: $event)] in 366
```

This error message gives no indication that the named argument is to blame. We should either consider supporting named arguments since they're an integral part of the language, or have an explicit error message.

While the latter sounds easier, simply identifying named parameters for error reporting would probably require most of the work necessary to support them altogether.

**Comment by matanlurey** [comment]
The current parser is (still) based on the intersection between TypeScript and Dart, unfortunately, and I think lack of named arguments is directly related to that (we have _some_ differences, like `?.` operator). 

I don't know how trivial it would be to support named arguments here. If we do end up supporting them, we should probably also make sure it is supported elsewhere (dependency injection, for example, where it is not).

**Comment by leonsenft** [comment]
@matanlurey Right I forgot we're not using `package:angular_ast` for expression parsing.

**Comment by matanlurey** [comment]
And we deleted all the code in `angular_ast` related to expression parsing :(

**Comment by leonsenft** [comment]
Yeah I remember now. We're going to be forced to implement something new anyway eventually. Perhaps we can address this issue then.

**Comment by matanlurey** [comment]
Added to the >5.0 milestone. We should try and track this (and fix) before 6.0.

**Comment by matanlurey** [comment]
So it looks like I could probably implement this...

I'll send a PoC out for review. We don't need to get it in though. The code isn't great, but it isn't worse than the existing parsing and output AST code (it basically just adds a few extra parameters to calls, and also will need a hack to solve for ambiguity around pipe syntax - which also uses `:` for arguments).
